### PR TITLE
Password strength gauge stimulus

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
This builds on #3219 addressing #3197 and #3223 

More importantly includes a refactor of the password gauge functionality to Stimulus.

There's a few decisions I made around the ability to compose components using JS "mixins" which I'm not sure it's "idiomatic Stimulus" if there is such a thing. Here's three captures of the end results in three areas:

- Reset password:
reset_password

- Register:
register

- Update password:
update_password

Hopefully it makes sense and we can move forwards with Stimulus using this pattern.

### CODE REVIEW

File: warehouse/static/js/warehouse/controllers/password_controller.js [hunk 1]
```diff
@@ -11,10 +11,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/* global zxcvbn */
+
```
Comment (di) [inline warehouse/static/js/warehouse/controllers/password_controller.js]: Accidental comment?

**Comment by di** [comment]
Also, thank you for tackling this!

File: warehouse/static/js/warehouse/controllers/password_controller.js (hunk 1 above)
Comment (yeraydiazdiaz) [reply warehouse/static/js/warehouse/controllers/password_controller.js]: `eslint` appeaser 😄

**Comment by yeraydiazdiaz** [comment]
I've updated this to a much simpler approach of adding multiple controllers to the form.

I feel like [some lines](https://github.com/pypa/warehouse/pull/3243/files#diff-7ae9dcecd75b93cd37df3ca32da2c081R224) are getting a big too wide to read comfortably. I'd be in favor of something like:

```
{{ form.password_confirm(
    placeholder="Confirm password",
    required="required",
    class_="form-group__input",
    data_target="password.password password-match.passwordMatch",
    data_action="keyup->password-match#checkPasswordsMatch") }}
```

Would we consider disabling the template linting rule around single line echo statements to allow that?

**Comment by di** [comment]
@yeraydiazdiaz I think that'd be fine.

**Comment by yeraydiazdiaz** [comment]
Seems it's not something we can disable in the `html_lint` command line 😞

**Pull request merged**